### PR TITLE
tools/check/ut: reduce the generated coverage file size

### LINKED ISSUE #32200: Generated coverage file too large cause uploading fail
## Enhancement

After using the new CI script https://github.com/pingcap/tidb/issues/30822
The generated coverage file become quite large (2G+), and fails to uploaded to codecov:

```
[2022-02-09T03:22:56.177Z] [2022-02-09T03:22:55.977Z] ['info'] Could not read coverage file (tidb.coverage): Error: There was an error reading the coverage file: RangeError [ERR_FS_FILE_TOO_LARGE]: File size (2913268767) is greater than 2 GB
```

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #32200

Problem Summary:

Before this change, the size of the generated coverage file is more than 2G.
It's too large to be uploaded to the platform.

### What is changed and how it works?

In the past, I simply combine all the coverage files of each test case to generate the final coverage file.
Now the coverage file is parsed and de-duplicated.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [X] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

File: tools/check/ut.go [hunk 1]
```diff
@@ -406,16 +409,123 @@ func collectCoverProfileFile() {
...
+		last := &input[len(input)-1]
+		if b.StartLine == last.StartLine &&
+			b.StartCol == last.StartCol &&
+			b.EndLine == last.EndLine &&
+			b.EndCol == last.EndCol {
+			if b.NumStmt != last.NumStmt {
+				panic(fmt.Errorf("inconsistent NumStmt: changed from %d to %d", last.NumStmt, b.NumStmt))
+			}
+			// Merge the data with the last one of the slice.
+			last.Count |= b.Count
```
Comment (hawkingrei) [inline tools/check/ut.go:494]: I think it should use ```+=```.

```
1 | 9 = 9
```

Comment (hawkingrei) [reply tools/check/ut.go:494]: it only has 1 or 0.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**